package.json

```diff
@@ -31,20 +31,25 @@
 		"postbuild": "cpy README.md ./dist && cpy package.json ./dist && cpy LICENSE ./dist",
 		"generate-all": "run-p \"generate:*\"",
 		"generate:configs": "ts-node tools/generate-configs",
-		"generate:rules-doc": "npm run build && npm run rule-doc-generator",
-		"format": "npm run prettier-base -- --write",
-		"format:check": "npm run prettier-base -- --check",
+		"generate:rules-doc": "pnpm build && pnpm rule-doc-generator",
+		"format": "pnpm prettier-base --write",
+		"format:check": "pnpm prettier-base --check",
 		"lint": "eslint . --max-warnings 0 --ext .js,.ts",
-		"lint:fix": "npm run lint -- --fix",
+		"lint:fix": "pnpm lint --fix",
 		"prepare": "is-ci || husky install",
 		"prettier-base": "prettier . --ignore-unknown --cache --loglevel warn",
 		"rule-doc-generator": "eslint-doc-generator --path-rule-list \"../README.md\" --path-rule-doc \"../docs/rules/{name}.md\" --url-rule-doc \"docs/rules/{name}.md\" dist/",
 		"semantic-release": "semantic-release",
 		"test": "jest",
 		"test:ci": "jest --ci --coverage",
-		"test:watch": "npm run test -- --watch",
+		"test:watch": "pnpm test -- --watch",
 		"type-check": "tsc --noEmit"
 	},
+	"engines": {
+		"node": "^12.22.0 || ^14.17.0 || >=16.0.0",
+		"pnpm": "^7.20.0"
+	},
+	"packageManager": "pnpm@7.26.2",
```

Comment:
This basically means "we support any pnpm greater or equal than v7.20.0 buy corepack will install v7.26.2 for you".